client/landing/gutenboarding/stores/domain-suggestions/resolvers.ts

```diff
@@ -7,10 +7,12 @@ import { apiFetch } from '@wordpress/data-controls';
 /**
  * Internal dependencies
  */
-import { DomainSuggestionQuery } from './types';
 import { receiveDomainSuggestions } from './actions';
 
-export function* getDomainSuggestions( queryObject: DomainSuggestionQuery ) {
+export function* __internalGetDomainSuggestions(
+	// Resolver has the same signature as corresponding selector without the initial state argument
+	queryObject: Parameters< typeof import('./selectors').__internalGetDomainSuggestions >[ 1 ]
```

Comment:
This assumes that there's only one arg to the selector aside from `state`, right? I.e. we're making an implicit assumption here. Would it be more stable to use `TailParameters` instead, which always _just_ discards the first arg and uses all the other ones?

Reply:
We can do nicer things, but not in `function*` generators 🙂 (as far as I know).

Here's the gist:

```ts
// Just prep, nothing to see here…
/**/ type TailParameters<F extends (a: any, ...args: any[]) => any> = F extends (a: any, ...args: infer T) => any ? T : never;
/**/ function selector(state: Record<string, any>, key: string) { }

// With a `const`, we can express this nicely as the type of the constant:
const resolver: (...args: TailParameters<typeof selector>) => void = (key) => console.log(key);

// That works with functions that can be defined as `function` or `=>`.
// `function*` (generators) do not have a `=>` version, so we don't have that option.

// That means we either make each parameter explicit:
function* resolver_(key: Parameters<typeof selector>[1]) {
    // We know what `key` is based on `selector` type.
}

// Or we do this sort of funny gymnastics, which makes them explicit anyways :)
function* resolver__(...[key]: TailParameters<typeof selector>) {
    // We know what `key` is based on `selector` type.
}
```

Reply:
We can do nicer things, but not in `function*` generators 🙂 (as far as I know).

Here's the gist:

```ts
// Just prep, nothing to see here…
/**/ type TailParameters<F extends (a: any, ...args: any[]) => any> = F extends (a: any, ...args: infer T) => any ? T : never;
/**/ function selector(state: Record<string, any>, key: string) { }

// With a `const`, we can express this nicely as the type of the constant:
const resolver: (...args: TailParameters<typeof selector>) => void = (key) => console.log(key);

// That works with functions that can be defined as `function` or `=>`.
// `function*` (generators) do not have a `=>` version, so we don't have that option.

// That means we either make each parameter explicit:
function* resolver_(key: Parameters<typeof selector>[1]) {
    // We know what `key` is based on `selector` type.
}

// Or we do this sort of funny gymnastics, which makes them explicit anyways :)
function* resolver__(...[key]: TailParameters<typeof selector>) {
    // We know what `key` is based on `selector` type.
}
```

Reply:
Oh wow :open_mouth: Thanks for explaining! I don't see at first glance _why_ we require `=>` syntax for this to work? :sweat_smile:

Reply:
> I don't see at first glance why we require => syntax for this to work?

We can provide a complete for a variable (constant):
```ts
// v~~~~ all declaration ~~~~ v
const f: ( a: string ) => string = 

// here comes the implementation
a => a;

// We don't have the option of typing `g` (the function name)
// independently of the implementation(to the best of my knowledge):
function g(a: string): string { return a; } 
```

Note that in former, we write a type like `( … ) => …`, but the function types its args and return in-place.

There's just no way to say "that function g has the type …" (again, maybe I'm missing something 🙂).